packages/edit-post/src/components/preferences-modal/general-preferences.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { useViewportMatch } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+import {
+	PreferencesModalSection,
+	PreferencesModalFeatureToggle,
+} from '@wordpress/interface';
+
+/**
+ * Internal dependencies
+ */
+import { EnablePublishSidebarOption } from './options';
+import { store as editPostStore } from '../../store';
+
+export default function GeneralPreferences() {
+	const isLargeViewport = useViewportMatch( 'medium' );
+	const showBlockBreadcrumbsOption = useSelect(
+		( select ) => {
+			const { getEditorSettings } = select( editorStore );
+			const { getEditorMode, isFeatureActive } = select( editPostStore );
+			const mode = getEditorMode();
+			const isRichEditingEnabled = getEditorSettings().richEditingEnabled;
+			const hasReducedUI = isFeatureActive( 'reducedUI' );
+			return (
+				! hasReducedUI &&
+				isLargeViewport &&
+				isRichEditingEnabled &&
+				mode === 'visual'
+			);
+		},
+		[ isLargeViewport ]
+	);
+
+	return (
+		<>
+			{ isLargeViewport && (
+				<PreferencesModalSection
+					title={ __( 'Publishing' ) }
+					description={ __(
+						'Change options related to publishing.'
+					) }
+				>
+					<EnablePublishSidebarOption
+						help={ __(
+							'Review settings, such as visibility and tags.'
+						) }
+						label={ __( 'Include pre-publish checklist' ) }
+					/>
+				</PreferencesModalSection>
+			) }
+
+			<PreferencesModalSection
+				title={ __( 'Appearance' ) }
+				description={ __(
+					'Customize options related to the block editor interface and editing flow.'
+				) }
+			>
+				<PreferencesModalFeatureToggle
+					scope="core/edit-post"
+					feature="reducedUI"
+					help={ __(
+						'Compacts options and outlines in the toolbar.'
+					) }
+					label={ __( 'Reduce the interface' ) }
+				/>
+				<PreferencesModalFeatureToggle
+					scope="core/edit-post"
+					feature="focusMode"
+					help={ __(
+						'Highlights the current block and fades other content.'
+					) }
+					label={ __( 'Spotlight mode' ) }
+				/>
```

Comment:
It's very similar to:

<img width="704" alt="Screen Shot 2021-08-23 at 14 59 50" src="https://user-images.githubusercontent.com/699132/130451617-57c0ea13-1cd5-42ff-89d5-fc292baed5f6.png">

It makes me wonder whether we should compose those components differently. What about we build a general-purpose feature toggle button and we use `as` prop to render it differently depending on the context:

```js
<EditorFeatureToggle
	as={ PreferencesModalToggle }
	scope="core/edit-post"
	feature="focusMode"
	help={ __(
		'Highlights the current block and fades other content.'
	) }
	label={ __( 'Spotlight mode' ) }
/>
```

Reply:
Hmm, yep, or a hook:
```jsx
const props = useFeaturePreference( 'core/edit-post', 'topToolbar' );
return (
    <PreferencesModalToggle
        label={ __( 'Top toolbar' ) }
        { ...props }
    />
);
```

Reply:
I had a look into it. I'm not sure it'd result in much gain, there would still be three components exported (`EditorFeatureToggle`, `MoreMenuToggle` and `PreferencesModalToggle`), which is more consistent, but I think it makes the API more complicated than a straightforward component that the implementer can import.

The problem with `as` is that it requires a component passed to it to have a particular interface (supporting props like `value` and `onChange`), so as implementers we end up writing adapter components to map those props to the underlying `MenuItem` (`isSelected` / `onClick`) and `ToggleControl` (`checked` / `onChange`) components.

A hook at least allows rewiring the props, but that can be verbose if there are lots of preferences (which is generally the case).

So I'm not sure it's worth reworking this.